cmd/collectors/ems/templating.go

```diff
@@ -126,8 +126,12 @@ func (e *Ems) ParseResolveEms(resolveEvent *node.Node, issueEmsProp emsProp) {
 
 	// check if resolved_key is present in template, if not then use the issue ems resolve key
 	if resolveKey = resolveEvent.GetChildS("resolve_key"); resolveKey == nil {
-		// resolving ems instance key is keys
-		prop.InstanceKeys = issueEmsProp.InstanceKeys[2:]
+		// IssuingEmsKey: index-messageName-bookendKey, ResolvingEmsKey would be bookendKey
+		if len(issueEmsProp.InstanceKeys) > 2 {
+			prop.InstanceKeys = issueEmsProp.InstanceKeys[2:]
+		} else {
+			e.Logger.Warn().Str("Ems name", issueEmsProp.Name).Msgf("Missing bookend keys")
```

Comment:
nope, in that case, prop.instanceKeys will be empty("").

Reply:
prop.instanceKeys is resolving ems' key

Reply:
index won't work as resolvingEms' index won't be available in issuingEms key. So, matching ems pair would not be possible based on index

Reply:
checking the same

Reply:
As per the `HasSuffix` definition, it will return true for empty substring. So, all the instances would be returned at `GetInstancesBySubkey` when input is "".

But, ems.go line 604, we do pass input as {issueEmsName+bookendKey}.
So, For example Lun.Offline pair, Even if Lun.online ems bookendKey is "", input would be `Lun.offline` and we can get all the lun.offline instances.

Due to missing bookendKey, all the lun.offline instances were resolved which is not right. 

Changing the default bookendKey as "index" with warning. non matching ems never resolve which is design.